src/txn/txn_rollback_to_stable.c

```diff
@@ -1526,6 +1529,15 @@ __rollback_to_stable_hs_final_pass(WT_SESSION_IMPL *session, wt_timestamp_t roll
 
     WT_TRET(__wt_session_release_dhandle(session));
 
+    /*
+     * Truncate history store entries from the partial backup remove list. The list holds all of the
```

Comment:
I had to think about this a bit. I guess it's guaranteed that RTS won't touch any of the tables *not* being restored, since it doesn't visit them because they aren't in the metadata. I can also see that this step needs to be here because it's kind of like post recovery cleanup. OTOH: it would have been nice not to have this special case in RTS.
That's a general tangent. My only point is that this is a piece of logic we'll need to watch out for, and ensure it remains correct and relevant, since it's not an obvious place to look for something associated with partial restores.